lang/rust/avro/src/types.rs

```diff
@@ -377,11 +377,16 @@ impl Value {
         }
     }
 
+    /// Validates the value against the provided schema.
+    ///
+    /// Arguments:
+    /// * `schema_resolution` - whether schema resolution rules should be applied when validating the `value`.
```

Comment:
Actually `types::resolve()` is supposed to be used for schema resolution. At least according to its Rustdoc.
So maybe `schema::find_schema_with_known_schemata()` wrongly uses `validate_internal()` ?!

Reply:
Yes - I considered this as well. If `UnionSchema::find_schema_with_known_schemata()` were not to use `validate_internal()`, then it would need another mechanism to select the right schema for a `types::Value`, given a set of schemas.

I suppose this could be done, but this seems like a larger lift than the current fix.

Reply:
I will take a deeper look in the coming days!